### DISCUSSION TITLE
Minor modification required

### DISCUSSION THREAD

**Discussion by hamiddhg** [jorgensd/dolfinx_mpc#79] [General]
First of all, thank you very much for this wonderful work. I just wanted to mention that in fine "bench_periodic.py" in line 147:  mpc.backsubstitution(uh) does not work since uh is a Function, not a vector. I simply changed it to mpc.backsubstitution(uh.vector) and it works. 
All the best,
Hamid,

**Comment by jorgensd** [comment]
This depends on what version of dolfinx_mpc you are running, as I Added this functionality earlier this year. Make sure you look at demos of the same version as your installation

**Reply by hamiddhg** [reply]
Thank you very much, Jorgen. Are there any citation preferences for dolfinx_mpc? I found the list on Fenics website but nothing about this work.

**Reply by jorgensd** [reply]
You can cite it as:
```
Dokken, J. S. (2022). DOLFINx-MPC (Version 0.6.0) [Computer software]
```
or with bibtex 
as
```
@software{Dokken_DOLFINx-MPC_2023,
author = {Dokken, Jørgen Schartum},
month = {feb},
title = {{DOLFINx-MPC}},
version = {0.6.0},
year = {2023}
}
```

**Reply by hamiddhg** [reply]
Thanks 🙏
